plugins/by-name/lazydev/default.nix

```diff
@@ -0,0 +1,134 @@
+{ lib, ... }:
+let
+  inherit (lib) types;
+  inherit (lib.nixvim)
+    defaultNullOpts
+    literalLua
+    mkNullOrOption'
+    nestedLiteralLua
+    ;
+
+  libraryType = types.submodule {
+    freeformType = with types; attrsOf anything;
+    options = {
+      files = mkNullOrOption' {
+        type = with types; listOf str;
+        example = [ "xmake.lua" ];
+        description = "A list of files that triggers the library to be loaded when required.";
+      };
+      mods = mkNullOrOption' {
+        type = with types; listOf str;
+        example = [ "wezterm" ];
+        description = "A list of modules that triggers the library to be loaded when required.";
+      };
+      path = lib.mkOption {
+        type = types.str;
+        example = "~/projects/my-awesome-lib";
+        description = "The path to the library.";
+      };
+      words = mkNullOrOption' {
+        type = with types; listOf str;
+        example = [
+          "wezterm"
+          "LazyVim"
+        ];
+        description = "A list of words that triggers the library to be loaded.";
+      };
+    };
+  };
+in
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "lazydev";
+  packPathName = "lazydev.nvim";
+  package = "lazydev-nvim";
+
+  maintainers = [ lib.maintainers.HeitorAugustoLN ];
+
+  description = ''
+    ### lazydev.nvim as a blink.cmp source
+
+    ```nix
+    {
+      plugins = {
+        lazydev.enable = true;
+
+        blink-cmp.settings = {
+          sources.providers = {
+            lazydev = {
+              name = "LazyDev";
+              module = "lazydev.integrations.blink";
+              # make lazydev completions top priority (see `:h blink.cmp`)
+              score_offset = 100;
+            };
+          };
+        };
+      };
+    }
+    ```
+  '';
+
+  settingsOptions = {
+    enabled =
+      defaultNullOpts.mkBool
+        (literalLua ''
+          function(root_dir)
+            return vim.g.lazydev_enabled == nil and true or vim.g.lazydev_enabled
+          end
+        '')
+        ''
+          Whether lazydev.nvim is enabled.
+        '';
+
+    integrations = {
+      cmp = defaultNullOpts.mkBool true ''
```

Comment:
We could benefit from adding some warnings when a user enables any of these and checking if they've enabled the corresponding module.

Reply:
Don't really think this is needed, `lazydev.nvim` only loads the integration if the module of what it is integrating is available

Reply:
I added warnings, since I thought more about it, and it makes sense.